crates/solver/src/driver.rs

```diff
@@ -283,52 +280,25 @@ impl Driver {
             }
         });
 
-        let previous_orders: Vec<OrderUid> = auction
-            .orders
-            .iter()
-            .map(|order| order.metadata.uid)
-            .collect();
         let balance_start = Instant::now();
-        let orders = self.order_balance_filter.filter(auction.orders).await;
+        let balances =
+            order_balance_filter::fetch_balances(self.balance_fetcher.as_ref(), &auction.orders)
+                .await;
```

Comment:
Fetching the balances before determining the auction deadline basically takes care of https://github.com/cowprotocol/services/issues/1461 as well. 👍